lighthouse-core/audits/byte-efficiency/unused-javascript.js

```diff
@@ -65,43 +71,128 @@ class UnusedJavaScript extends ByteEfficiencyAudit {
     }
 
     return {
+      unusedByIndex,
       unusedLength: unused,
       contentLength: maximumEndOffset,
     };
   }
 
   /**
-   * @param {Array<{unusedLength: number, contentLength: number}>} wasteData
-   * @param {LH.Artifacts.NetworkRequest} networkRecord
+   * @param {LH.Audit.ByteEfficiencyItem} item
+   * @param {WasteData[]} wasteData
+   * @param {LH.Artifacts.Bundle} bundle
+   * @param {ReturnType<typeof UnusedJavaScript.determineLengths>} lengths
+   */
+  static createBundleMultiData(item, wasteData, bundle, lengths) {
+    if (!bundle.script.content) return;
+
+    /** @type {Record<string, number>} */
+    const files = {};
+
+    const lineLengths = bundle.script.content.split('\n').map(l => l.length);
+    let totalSoFar = 0;
+    const lineOffsets = lineLengths.map(len => {
+      const retVal = totalSoFar;
+      totalSoFar += len + 1;
+      return retVal;
+    });
+
+    // @ts-ignore: We will upstream computeLastGeneratedColumns to CDT eventually.
+    bundle.map.computeLastGeneratedColumns();
+    for (const mapping of bundle.map.mappings()) {
+      let offset = lineOffsets[mapping.lineNumber];
+
+      offset += mapping.columnNumber;
+      const lastColumnOfMapping =
+        // @ts-ignore: We will upstream lastColumnNumber to CDT eventually.
+        (mapping.lastColumnNumber - 1) || lineLengths[mapping.lineNumber];
+      for (let i = mapping.columnNumber; i <= lastColumnOfMapping; i++) {
+        if (wasteData.every(data => data.unusedByIndex[offset] === 1)) {
+          // @ts-ignore
+          files[mapping.sourceURL] = (files[mapping.sourceURL] || 0) + 1;
+        }
+        offset += 1;
+      }
+    }
+
+    const transferRatio = lengths.transfer / lengths.content;
+    const topUnusedFilesSizes = Object.entries(files)
+      .filter(d => d[1] * transferRatio >= 1024)
+      .sort((a, b) => b[1] - a[1])
```

Comment:
can you also destructure in the filter/sort like you did in the map()

feels like the var names are `sourceURL` and `unusedBytes` right?